### ISSUE TITLE
travis: The setpgid_kill test is unstable

### ISSUE THREAD

**Issue by SimonKagstrom** [SimonKagstrom/kcov#191]
Sometimes hangs, probably waiting for a result which has already arrived or so.

**Comment by ishbguy** [comment]
I encounter the same situation then running kcov after  travis build, kcov hang without output for a long time, and finally killed by travis. For details you can see: https://travis-ci.org/ishbguy/baux/builds/376708953

**Comment by SimonKagstrom** [comment]
OK, interesting - and this is with bash as well. The `setpgid_kill` test has been stable for a while though, so it appears like I've managed to avoid a race somehow.

Does this happen if you run it on your computer as well?

**Comment by ishbguy** [comment]
yes, I encounter the same situation when running `kcov` on my local machine , so I need to kill kcov process to get back my shell controling. I try to use `--bash-method=DEBUG` option, `kcov` can run through, but the coverage report shows a **low percentage (lower than 10%)** on my code, which I think `kcov` collects not enough datas.

**Comment by SimonKagstrom** [comment]
Yes, sounds like that. Can you run with --debug=31 and see if you can see
where it hangs? Alternatively, run kcov via gdb (gdb --args kcov [other
options]) and break it when it hangs)

**Comment by ishbguy** [comment]
I have tried to run `kcov --debug=31`, and this is the out put:
[kcov-debug.txt](https://github.com/SimonKagstrom/kcov/files/1992192/kcov-debug.txt)

**Comment by ishbguy** [comment]
I use `lldb` to debug, and I was hanged in the `handleStdout`'s `getline` calling, a part of the debug output is in the txt below:
[lldb-kcov-debug.txt](https://github.com/SimonKagstrom/kcov/files/1992515/lldb-kcov-debug.txt)

**Comment by SimonKagstrom** [comment]
Hmm.. That's the one which reads from the covered programs output. Do you need to use `--bash-method=DEBUG`, and does it make a difference without it?

Does running bash with `-x` work as expected?

**Comment by ishbguy** [comment]
1. running `kcov --bash-method=DEUBG dir test-script.sh` is OK, but `REPORT hit` little code.
2. running `kcov dir test-script.sh` will hang.
3. running `bash -x test-script.sh` is OK.

**Comment by SimonKagstrom** [comment]
OK! Thanks for the analysis, this should be not too difficult to fix! Perhaps a pull-request even?

So does your program output stuff to stdout without newlines? Anyway, I wrote #248 to cover this, since this part of the code is bash-specific and the original bug describes a problem which is present in binary files only.
